modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,371 @@
+
+      expect(value1).to.equal(value2);
+      expect(value2).to.equal(value3);
+    });
+
+    it('TINY-11974: Should handle invalid user IDs gracefully', async () => {
+      const editor = hook.editor();
+      const invalidIds = [ '', ' ', undefined, null, false, 123 ] as any[];
+
+      // Test each invalid ID asynchronously
+      await Promise.all(Arr.map(invalidIds, (invalidId) => {
+        const [ promise ] = editor.userLookup.fetchUsers([ invalidId ]);
+        return expect(promise).to.be.rejectedWith(
+          `User ${invalidId} not found`,
+          `Should reject for invalid ID: ${invalidId}`
+        );
+      }));
+    });
+
+    it('TINY-11974: Should handle large batches of users efficiently', async () => {
+      const editor = hook.editor();
+      const userIds = Arr.range(100, (i) => `test-user-${i}`);
+
+      const promises = editor.userLookup.fetchUsers(userIds);
+      const users = await Promise.all(promises);
+
+      await Promise.all(Arr.map(users, async (user, index) =>
+        expect(Promise.resolve(user)).to.eventually.have.property('id').that.equals(`test-user-${index}`)
+      ));
+    });
+
+    it('TINY-11974: Should handle various combinations of optional properties correctly', async () => {
+      const editor = hook.editor();
+      const testCases = [
+        {
+          input: { id: 'user-1' },
+          expected: { id: 'user-1' }
+        },
+        {
+          input: { id: 'user-2', name: 'John' },
+          expected: { id: 'user-2', name: 'John' }
+        },
+        {
+          input: {
+            id: 'user-3',
+            name: undefined,
+            avatar: null,
+            description: '',
+            custom: {}
+          },
+          expected: { id: 'user-3', description: '', custom: {}}
+        },
+        {
+          input: {
+            id: 'user-4',
+            name: 'Jane',
+            avatar: 'avatar.jpg',
+            description: null,
+            custom: { role: 'admin' }
+          },
+          expected: {
+            id: 'user-4',
+            name: 'Jane',
+            avatar: 'avatar.jpg',
+            custom: { role: 'admin' }
+          }
+        }
+      ];
+
+      // Override the fetch_users fn to return our test cases
+      editor.options.set('fetch_users', () => Promise.resolve(Arr.map(testCases, (c) => c.input)));
+
+      const userIds = Arr.map(testCases, (c) => c.input.id);
+      const promises = editor.userLookup.fetchUsers(userIds);
+
+      await expect(Promise.all(promises)).to.eventually.deep.equal(
+        Arr.map(testCases, (c) => c.expected),
+        'Should resolve with expected user objects'
+      );
+    });
+  });
+
+  context('fetchUsers with override', () => {
+    let editor: Editor;
+    let originalFetchUsers: (userIds: string[]) => Promise<User[]>;
+
+    beforeEach(() => {
+      editor = hook.editor();
+      originalFetchUsers = editor.options.get('fetch_users');
```

Comment:
This can be stored in a single `before` function at the top of the file. The value we want to restore it to after every test in the file is the value it was before any of these tests run.